add planning section (#46)
talis/talis.github.io: topics/planning.md

---
layout: site
title: Planning
body_class: planning
---

### Planning repository

> Each sub team within the Engineering team should have a single planning repository and Zenhub board to track their work.  If a sub team creates a separate board or github project then that single view of the sub team is lost.  An exemption to this is ramp up stories for onboarding new team members, these issues should not be on the critical path and are not estimated.  Therefore they have no impact on the teams progress or velocity.

Reviewer: ```suggestion
Each sub-team within the Engineering team should have a single planning repository and Zenhub board to track their work.  If a sub-team creates a separate board or GitHub project then that single view of the sub-team is lost.  An exemption to this is ramp up stories for onboarding new team members: these issues should not be on the critical path and are not estimated, therefore they should have no impact on the teams progress or velocity.
```

### Estimating

> With team velocity being a good indicator of what a team can achieve in a sprint, a common understanding of an estimate on a story provides a quick view benefit and helps if this is common across the teams.  The rule of thumb for the following points within the fibonacci range is:

Reviewer: ```suggestion
With team velocity being a good indicator of what a team can achieve in a sprint, a common understanding of an estimate on a story provides a quick-view benefit and helps if this is common across the teams.  The rule of thumb for the following points within the Fibonacci range is:
```

|Points|Days|
|------|----|
|5     |1|
|21 |5|
|40 |no idea, break up / > 10 days)

> >Note: Person time here is only to provide an idea, we can not help but think about how long it would take us to do a task.  A reason for using points is really for us to gauge the complexity of a task, and there are usually other time based factors that surround a piece of work from being complete.  When a team thinks and estimates this way, it is natural and expected that velocity is less than the calculated person days.

Reviewer: I think this view that we cannot help but think of time is not actually true - that's exactly what other people do with sprint point estimates - time-equivalence is a no-no in scrum, and a debate will follow.

Rather than saying that we cannot help it, we should acknowledge that we are deliberately choosing to use a point-to-time-equivalent because that suits us better in practice. Then, state how that works.

> >Estimates should never be altered once a sprint starts.

Reviewer: I think we should state that we should record actual person-time in point-equivalence where we introduce the term "triage" (or its equivalent) below.

### Project and Theme work
Zenhub Projects and Epics should be used to track issues against a theme of work.  Examples of this are Accessibility, Kubernetes Migration, Ebook Central Integration, Mobile Annotations etc. Epics provide an overview of how a theme of work is progressing, it is not a complete picture from inception but should represent the overall effort and timeline of a project.
Mutually exclusive reporting labels

> Labels are a good way for us to distinguish and filter on our issues, we can also use them to report on where we are spending our effort.  In order to track effort, every issue should have a mandatory mutually exclusive label.

Reviewer: ```suggestion
Labels are a good way for us to distinguish and filter on our issues, we can also use them to report on where we are spending our effort.  In order to track effort, every issue should have a mandatory, mutually-exclusive label from the set below:
```

|Label|Description|
|-----|-----------|

> |Triage|Issues that are related to support requests, pager duty alerts, broken builds/dependencies, anything not theme related but needs to be done within the current sprint.|

Reviewer: We need to tighten this up, I think - it's the controversial one.

First, "not theme related" doesn't seem to fit with "non-theme".

Second, "triage" has a very particular real-world meaning - it's about getting an understanding of a problem, not about fixing the problem. When we perform triage, the likelihood is that we'll understand a problem, but that leads to the next question: is the work to *fix* the problem labelled as "triage", or something else? We need to give some clear guidance on this distinction. If it goes into a subsequent sprint, it's clearly not triage; but if it's urgent enough to be done in the same sprint...? That not "triage" by the meaning of the word, but should it get the "triage" label?

Yes, this is a semantic argument. Perhaps an entirely new term would help us side-step the issues (because we already have "triage" processes, and work that is not previously planned is naturally "unplanned" which we have elsewhere).

So either we use a new and non-overloaded term ("unexpected" or "interruption" or "disruption" or "surprise!" etc.) and give it a clear meaning, or we give some very explicit support guidance here to help people understand the labelling (or both).

Reviewer: I'd also start with the others - they're all clearly defined. Leave this term until last in the list.

Reviewer: Basically, we want to differentiate these two:

"stuff that came in without warning, that was not related to the theme or non-theme work already in the sprint, but which occupied our time" (currently "triage")

and:

"stuff we uncovered as doing our theme or non-theme work, that needed to be fixed to progress the theme or non-theme work" (not "triage")

Author: Updated

---
layout: site
title: Planning
body_class: planning
---

### Planning repository

> Each sub-team within the Engineering team should have a single planning repository and Zenhub board to track their work.  If a sub-team creates a separate board or Github project then that single view of the sub-team is lost.  An exemption to this is ramp up stories for onboarding new team members: these issues should not be on the critical path and are not estimated, therefore they should have no impact on the teams progress or velocity.

Reviewer 2: We have used Github projects before and found them useful, for example on the accessibility project, to communicate to Product/UX about what needs attention by which team - https://github.com/talis/rapprochement/projects/4 - and for managing a backlog of tech debt issues. I presume those usages are still permissible

Author: @camillef yes I think that can be reworded to include those uses.  I think there is only a problem when Github projects have issues that are being worked on that are not included in the sprint (except onboarding).

### Estimating
With team velocity being a good indicator of what a team can achieve in a sprint, a common understanding of an estimate on a story provides a quick view benefit and helps if this is common across the teams.  The rule of thumb for the following points within the Fibonacci range is:

|Points|Days|
|------|----|
|5     |1|
|21 |5|
|40 |No idea, break down story / > 10 days)

The number of days derived from points assigned to an issue is not neccesarily how long it will take to deliver that issue.  We choose to use time-to-points because in practice that suits us better when thinking about a task at hand.  In reality there are usually other time based factors that surround a piece of work from being completed.  When a team thinks and estimates this way, it is natural and expected that velocity is less than the total time available to that team in a sprint.

> **Estimates on planned stories should never be altered once a sprint starts.**  Issues that are introduced into the current sprint with the `Triage` label are allocated points for the actual time it takes to complete.

Reviewer: ```suggestion
**Estimates on planned stories should never be altered once a sprint starts.**  Issues that are introduced into the current sprint with the `Triage` label are allocated points for the actual person-time it takes to complete.
```

It's not lapse time; it's how much effort we spent.

> **Estimates on planned stories should never be altered once a sprint starts.**  Issues that are introduced into the current sprint with the `Triage` label are allocated points for the actual time it takes to complete.  All other issues derived from Theme or Non-theme work are estimated accordingly.

Reviewer: ```suggestion
**Estimates on planned stories should never be altered once a sprint starts.**  Issues that are introduced into the current sprint with the `Triage` label are allocated points for the actual person-time it takes to complete.  All other issues derived from Theme or Non-theme work are estimated accordingly.
```

> **Estimates on planned stories should never be altered once a sprint starts.**  Issues that are introduced into the current sprint with the `Triage` label are allocated points for the actual person-time it takes to complete.  All other issues derived from Theme or Non-theme work are estimated accordingly.

Reviewer 2: In practice, this happens fairly regularly. We also might pull forwards stories from future sprints and add an estimate on-the-fly for those. Is this really such a hard and fast rule?

Author: Is it regular practice to change estimates for issues within a current sprint? Adding an estimate to pull forwards is fine if they do not already have one.

Reviewer: We shouldn’t be working anything without an estimate - if a pull forward doesn’t have an estimate, it needs one before being scheduled.

Reviewer 2: Okay, thanks. I misunderstood estimates being _altered_ to include _adding an estimate_

Reviewer: I wonder if it’s easy to see how often we change estimates...?

Author: > I wonder if it’s easy to see how often we change estimates...?

Don't think the value is tracked in the history of the issue.  Just thought maybe a section on timebox would be good in here.

Reviewer: I think it does show you on an individual issue; not sure if you can see it in aggregate though. Might be worth a dig because that might help us understand (a) if it is common; (b) if it’s actually a good idea to change from estimate to actual on completion (that’s only true if it’s easy to see the numbers in aggregate)

### Project and Theme work

> Zenhub and Epics should be used to track issues against a theme of work.  Examples of this are Accessibility, Kubernetes Migration, Ebook Central Integration, Mobile Annotations etc. Epics provide an overview of how a theme of work is progressing, it is not a complete picture from inception but should represent the overall effort and timeline of a project.

Reviewer 2: ```suggestion
Zenhub Epics should be used to track issues against a theme of work.  Examples of this are Accessibility, Kubernetes Migration, Ebook Central Integration, Mobile Annotations, etc. Epics provide an overview of how a theme of work is progressing. It is not a complete picture from inception but should represent the overall effort and timeline of a project.
```

### Mutually exclusive reporting labels

> Labels are a good way for us to distinguish and filter on our issues, we can also use them to report on where we are spending our effort.  In order to track effort, every issue should have a mandatory mutually-exclusive label from the set below:

Reviewer 2: ```suggestion
Labels are a good way for us to distinguish between and filter our issues. We can also use them to report on where we are spending our effort.  In order to track effort, every issue should have a mandatory mutually-exclusive label from the set below:
```

|Label|Description|
|-----|-----------|

> |Theme|Issues related to theme work, even those created within a current sprint related to that theme.|

Reviewer 2: Can we be a little clearer about what "theme work" actually means. I've understood it in the Elevate team to be "anything from Product", including one-off functionality or UI bug fixes, or pieces that fall under "Continuous Improvement" (is this a theme in itself?) and "non-theme" work to be anything is motivated by the tech team rather than a product request. Is this understanding correct?

Reviewer 2: We might have tech themes like "Improve build stability" but I'm guessing this doesn't count as a "theme" here...?

Author: @camillef good point. I'll add a definition.

Author: @camillef are the labels better described now?

> |Theme|Theme work is specific goal focused project work that at a future date will be completed.  That goal can be a product feature or technical strategy.  A single theme usually maps 1:1 with an Epic.  Mobile annotations or supporting LTI 1.3 would be a theme, but something like Continuous Improvment or adhoc technical debt would not.|

Reviewer: ```suggestion
|Theme|Theme work is specific goal focused project work that at a future date will be completed.  That goal can be a product feature or technical strategy.  A single theme usually maps 1:1 with an Epic.  Mobile annotations or supporting LTI 1.3 would be a theme, but something like Continuous Improvement or ad hoc technical debt would not.|
```

> |Non-theme|Issues that are not related to theme work but are planned in accordingly, even those created from existing planned work within a current sprint.  Examples are Continous Improvement and technical debt.|

Reviewer 2: I'm not sure I understand "even those created from existing planned work within a current sprint" - are these issues that enter the current sprint as a spin-off of theme work in the current sprint?

Author: Yes.  Working on issue A gets blocked, create issue B and estimate (as we want to continue the work rather than wait for the next sprint), complete and continue on issue A.

Reviewer: The general idea is that we want to spot when we are being heavily disrupted by unexpected issues. To achieve this, we’d like to distinguish “this is work we planned” from “this is work that arrived without warning”.

The spirit of this suggestion is to avoid people discovering they underestimated some planned work, and then marking it as “triage”. It’s really an effort to distinguish it from “triage” - if planned work expanded, that’s not triage, it’s more planned work. (If we get lots of these created during a sprint, we have a different problem.)

Reviewer 2: Okay, so there is perhaps a little subtlety there. Techincal debt fixes that arise from "theme" work might be labelled as "non theme", although they might serve to help us deliver the "theme"

Reviewer 2: Oops, commented before seeing your comment @junglebarry. That's really useful context, thanks.

Author: @camillef I took your comment as general to non-theme or theme, and not specific to non-theme.  I'll clarify.

|10%|Personal development tasks.|

> |Triage|This is applied **only** to issues that enter the current sprint, and that are not created due to work currently in the sprint.  Examples include: support requests, pager duty alerts, fixing broken builds etc. At the start of a sprint no issues should carry the Triage label.|

Author: @rsinger i've talked to Laura and Zena and they are happy to rename the current triage labels to zendesk* for tracking what comes through support.
